vmm/src/vm.rs

```diff
@@ -2115,14 +2115,28 @@ impl Vm {
         let mem = guest_memory.memory();
 
         for range in ranges.regions() {
-            mem.read_exact_from(GuestAddress(range.gpa), fd, range.length as usize)
-                .map_err(|e| {
-                    MigratableError::MigrateReceive(anyhow!(
-                        "Error transferring memory to socket: {}",
-                        e
-                    ))
-                })?;
+            let mut offset: u64 = 0;
```

Comment:
Add a comment saying vm-memory is buggy and pointing to the bug you opened with vm-memory about the implementation so that in the future we don't switch this over to the correct looking API when it is in fact buggy.

Reply:
Will do!